src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -1520,10 +1545,26 @@ def to_series(self) -> None:
         Index.to_frame : Convert an Index to a DataFrame.
         Series.to_frame : Convert Series to DataFrame.
         """
-        # TODO: SNOW-1458117 implement to_series
+        from snowflake.snowpark.modin.pandas import Series
 
-    @index_not_implemented()
-    def to_frame(self) -> None:
+        if name is None:
```

Comment:
the to_series and to_frame seems done in a very similar way, where we can do 
1) reset the index to to make the original column a data column, and it will also create a default index that is row_number
2) set_index if needed, the index can be a given index or original index for series, and the original index or None for dataframe
3) correct the data column name as post processing, for series, we need to set the result series name, for dataframe, need to rename the column if name is given.
Also, you might want to pay attention to the final name, it seems if the original index has no name, and no name is given, pandas gives a number label 0 as pandas label for the data column.
Step 1 and 2 can be extracted as a common function used by both series and dataframe

Reply:
I don't think I can do step 2 in a common function because we want to set index pretty differently in both cases

Reply:
I believe both `to_series` and `to_frame` are almost the same right? We only need to handle name differently

Reply:
No, `to_frame` sets the index as the existing index or a default index, it does not allow users to input new index data. `to_series` allows users to input index data, that is why there is one join for tests that include passing an index